tests/test_authentication.py

```diff
@@ -140,13 +139,17 @@ def test_unknown_creds(self):
                               '^access denied: CredentialsLookupError: ')
 
     def test_alternative_credential_lookup(self):
+        # Temporarily use alternate auth
+        old_auth = self.auth
+        self.auth = AlternateAuthentication()
+
         sender = self._sender(credentials=ALTERNATIVE_CREDS)
         req = self._request(sender)
-        lookup_path = "%s.alternative_lookup" % __name__
-        with self.settings(HAWK_CREDENTIALS_LOOKUP=lookup_path):
```

Comment:
It's [a call to import_module](https://github.com/django/django/blob/6e55e1d88a5c4453e25f0caf7ffb68973de5c0ba/django/utils/module_loading.py#L7), just like under the hood of the import statement (ultimately calling `__import__`). I'd rather optimize for testability since without it we won't be able to cover this logic. The `reload_module` alternative is too intrusive because test runners (and other things) tend to get easily confused when you switch module references.